builder/package.centos-7

```diff
@@ -59,6 +59,7 @@ fpm \
   -d xz-devel \
   -d zip \
   -d zlib-devel \
+  -d java-11-openjdk-devel \ 
```

Comment:
Since Java is often an optional dependency and somewhat heavy, could we leave it out of the DEB/RPM package dependencies? That would better support users who want to switch between Java 8, 11, 17, etc. as well. I think the default R packages from the various distros also leave Java as an optional dependency.

Reply:
We can leave java out but why do we then spend effort on configuring Java in the first place ?

Reply:
I'm not sure if there was a specific reason for configuring Java in the first place, and would guess that it was just borrowed from how Debian and Fedora packaged R. It is nice to be able to just install Java and have it work out of the box without `R CMD javareconf`. But I still do think Java is too heavy to add as a DEB/RPM dependency since these are billed as "minimal" R builds and Java is an optional dependency. It'll add a few hundred MBs to many consumers who don't necessarily need Java, like the RStudio R Docker images, the RStudio Pro Product images, the GitHub Actions for R, and others. And those on Java 8 or 17 will have an extra unused installation of Java.

Reply:
Ok, I definitely agree with your sentiment about the size of the R build if we go with the Java integration.

Reply:
Yeah, it's kind of both opinionated and minimal. We do some opinionated things that aren't completely minimal, like including OpenBLAS and the `-devel` packages of a few libraries that are required to install packages that link against the R shared library (which would only be included in `r-base-dev`, not `r-base` for example). A separate DEB/RPM that includes Java would be nice, but agreed that it'd complicate installation. Maybe a middle ground would be to provide users with a preconfigured Java where possible, like in the various Docker images. That article also looks great, and I'll try to reference it when any Java configuration questions come up.

Do you still want to explicitly set `JAVA_HOME` to at least make a first time Java installation more seamless? I've done that with the RHEL 9 R builds: https://github.com/rstudio/r-builds/pull/133/files#diff-7c6c557e074afff5ac851dbf63d384a5deaea84626a0f6945f41767510c08c39R77-R83. I think it's still better than letting R default it to something like `/usr/lib/jvm/java-11-openjdk-11.0.16.0.8-1.el9_0.x86_64`.

Reply:
Yes, setting `JAVA_HOME` for the RHEL9 builds to the more generic folder I would strongly recommend. Also agree with the notion of extra DEB/RPM packages for Java where we even could be in the business of providing packages for any JDK offered, e.g. 1.8.0, 11, 17, ...